metrics_api/lib/opentelemetry/metrics/meter.rb

```diff
@@ -148,20 +158,29 @@ def create_observable_gauge(name, unit: nil, description: nil, callbacks: nil)
       #
       # @return [Instrument::ObservableUpDownCounter]
       def create_observable_up_down_counter(name, unit: nil, description: nil, callbacks: nil)
-        create_instrument(:observable_up_down_counter, name, unit, description, nil, callbacks) { NOOP_OBSERVABLE_UP_DOWN_COUNTER }
+        register_instrument(name, NOOP_OBSERVABLE_UP_DOWN_COUNTER)
+      end
+
+      # @api private
+      def each_instrument
+        @mutex.synchronize do
+          @instrument_registry.each do |name, instrument|
+            yield(name, instrument)
+          end
+        end
       end
 
       private
 
-      def create_instrument(kind, name, unit, description, advice, callbacks)
+      def register_instrument(name, instrument)
         name = name.downcase
 
         @mutex.synchronize do
           if @instrument_registry.include?(name)
             OpenTelemetry.logger.warn("duplicate instrument registration occurred for #{name}")
           end
 
-          @instrument_registry[name] = yield
+          @instrument_registry[name] = instrument
         end
       end
```

Comment:
Renamed `create_instrument` to `register_instrument` for clarity and I couldn't take advantage of using a block, so I've changed it to receive instrument as an argument.